### PR TITLE
Feature/NQ-1147: RegModel based regression test suite

### PR DESCRIPTION
**Pull request by Jonah-Foley**
Previous PR for this ticket was named wrong, this is the new version with the correct branch name. 

YouTrack ticket: https://nu-quantum.youtrack.cloud/issue/NQ-1147/Utilise-RegModel-to-create-regression-test-suite

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,4 +2,6 @@ peakrdl
 mako
 pre-commit
 numpy
-cocotb
\ No newline at end of file
+cocotb
+cocotb-bus
+pytests
```
Comment (ghost) [inline requirements.txt]: This should be `pytest`?

**Review by ghost — CHANGES_REQUESTED** [review]
Think it's just that one typo and it can go.

File: requirements.txt (hunk 1 above)
Comment (Jonah-Foley) [reply requirements.txt]: Fixed, there were a few other things added too:
- in `regressions/test.py` corrected handling of environment variables if they werent set by the user
- Corrected active/inactive reset value in `regressions/testbench.reset()` (previously hard coded despite the logic being there to deduce active high/low form the reg map)

**Review by ghost — DISMISSED** [review]
LGTM

**Review by ghost — APPROVED** [review]
LGTM.

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — DISMISSED** [review]

**Comment by ghost** [comment]
@Jonah-Foley I'm totally lost - I've approved this review multiple times but it's still showing as "Review required".  WTF?

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
I removed a branch protection setting that seems to cause issues.  Not really sure of the reasons but it's in now.

**Pull request merged**